libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -120,6 +122,53 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg, const uint8_t i
     msg.position_covariance[8] = vdopSq;
 }
 
+#include "generated/TransformStamped.h"
+
+void AP_DDS_Client::populate_static_transforms(tf2_msgs_msg_TFMessage& msg)
+{
+    msg.transforms_size = 0;
+
+    auto &gps = AP::gps();
+    //! @todo Should we only publish the enabled GPS receivers, rather than the max amount?
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        const auto gps_type = gps.get_type(i);
+        if (gps_type == AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+            continue;
+        }
+        update_topic(msg.transforms[i].header.stamp);
+        char gps_frame_id[16];
+        //! @todo should GPS frame ID's be 0 or 1 indexed in ROS?
+        sprintf(gps_frame_id, "GPS_%u", i);
```

Comment:
please use snprintf()

Reply:
Solved in cd152031793694a27ccd01286fa5c981fbbd9f6a